tests/glitch/test_main.py

```diff
@@ -15,28 +15,15 @@ def setUp(self):
         self.loop = asyncio.get_event_loop()
         self.context = mock.Mock()
         self.rule = mock.Mock()
+        self.rule.log.debug = print
```

Comment:
This whole test is really an integration test, and is getting increasingly unwieldy. Replacing it with a matrix test plan.